### ISSUE TITLE
project (data-lovers): issue with "Rick & Morty" dataset

### ISSUE THREAD

**Issue by JulianaAmoasei** [Laboratoria/bootcamp#898]
During the Data Lovers project we came across a problem with Rick & Morty dataset. This set uses the Rick & Morty API links to get the character's pics (example, "https://rickandmortyapi.com/api/character/avatar/1.jpeg"), and apparently the API has a daily (ou 12h) request limit. We found out about this when the students started receiving error messages when trying to show the character's pics on the browser.

As a quick fix, I duplicated the dataset replacing the images with placeholders of the same dimensions (from http://placekitten.com/, of course 🐱), so the students could keep working. 

Anyways, this is a popular show, the data is nicely structured for the project's purposes (at least in my opinion) and many students chose to work with it. Accepting ideas on how we should deal with this request limit issue next time!

**Comment by diegovelezg** [comment]
Related: https://github.com/Laboratoria/bootcamp/issues/873

**Comment by juanjordan** [comment]
Hi! So I read this issue a few days ago and came up with the idea to host those images in our own Laboratoria org account. I ran this script with some variations given the request limit but it worked:
```
const request = require('request');
const fs = require('fs');
const rmData = require('./rickandmorty.json').results;

const success = name => {
  console.log('name', name);
};

const error = error => {
  console.log('error', error);
};

const images = rmData.map(character => new Promise((resolve, reject) => {
  const name = character.image.split('/')[6];
  request(character.image).pipe(fs.createWriteStream(`./images/${name}`))
  .on('close', resolve(() => success(name)))
  .on('error', reject(error));
}));

const start = () => Promise.all(images.map(promise => promise.catch(error => null)));

start();
``` 

now we can find those images in https://github.com/Laboratoria/rick-and-morty-images
and to hot link any of them I think we can use the raw url. e.g.: https://raw.githubusercontent.com/Laboratoria/rick-and-morty-images/master/images/1.jpeg

If you all are ok with this I think we can update the data set so it'd use our own self-hosted images.

Peace among worlds!
![](https://raw.githubusercontent.com/Laboratoria/rick-and-morty-images/master/images/1.jpeg)

**Comment by JulianaAmoasei** [comment]
awesoooome <3

**Comment by juanjordan** [comment]
Just an update, in GDL004 we're using this new image repository. We'll let you know how it goes and from there we can send a PR to master.

**Comment by lupomontero** [comment]
Nice one @juanjordan!

Quedamos a la espera de PR... :pray:

**Comment by lupomontero** [comment]
@juanjordan: any news WRT this issue?

**Comment by juanjordan** [comment]
Hola! We just finished the Data Lovers project in GDL004 and we saw no problem using our own backup repository. Yesterday @contrerasjf0 sent a PR regarding this issue #920. 
So I think we can safely update our data set. 🚀

**Comment by lupomontero** [comment]
Closed by #920
